backend/sirius-components-collaborative-formdescriptioneditors/src/main/java/org/eclipse/sirius/components/collaborative/formdescriptioneditors/handlers/DeleteWidgetEventHandler.java

```diff
@@ -0,0 +1,104 @@
+
+    public DeleteWidgetEventHandler(IObjectService objectService, ICollaborativeFormDescriptionEditorMessageService messageService, MeterRegistry meterRegistry) {
+        this.objectService = Objects.requireNonNull(objectService);
+        this.messageService = Objects.requireNonNull(messageService);
+
+        // @formatter:off
+        this.counter = Counter.builder(Monitoring.EVENT_HANDLER)
+                .tag(Monitoring.NAME, this.getClass().getSimpleName())
+                .register(meterRegistry);
+        // @formatter:on
+    }
+
+    @Override
+    public boolean canHandle(IFormDescriptionEditorInput formDescriptionEditorInput) {
+        return formDescriptionEditorInput instanceof DeleteWidgetInput;
+    }
+
+    @Override
+    public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, IFormDescriptionEditorContext formDescriptionEditorContext,
+            IFormDescriptionEditorInput formDescriptionEditorInput) {
+        this.counter.increment();
+
+        String message = this.messageService.invalidInput(formDescriptionEditorInput.getClass().getSimpleName(), DeleteWidgetInput.class.getSimpleName());
+        IPayload payload = new ErrorPayload(formDescriptionEditorInput.getId(), message);
+        ChangeDescription changeDescription = new ChangeDescription(ChangeKind.NOTHING, formDescriptionEditorInput.getRepresentationId(), formDescriptionEditorInput);
+
+        if (formDescriptionEditorInput instanceof DeleteWidgetInput) {
+            String widgetId = ((DeleteWidgetInput) formDescriptionEditorInput).getWidgetId();
+            boolean deleteWidget = this.deleteWidget(editingContext, formDescriptionEditorContext, widgetId);
+            if (deleteWidget) {
+                payload = new DeleteWidgetSuccessPayload(formDescriptionEditorInput.getId());
+                changeDescription = new ChangeDescription(ChangeKind.SEMANTIC_CHANGE, formDescriptionEditorInput.getRepresentationId(), formDescriptionEditorInput);
+            }
+        }
+
+        payloadSink.tryEmitValue(payload);
+        changeDescriptionSink.tryEmitNext(changeDescription);
+    }
+
+    protected boolean deleteWidget(IEditingContext editingContext, IFormDescriptionEditorContext formDescriptionEditorContext, String widgetId) {
+        var optionalSelf = this.objectService.getObject(editingContext, formDescriptionEditorContext.getFormDescriptionEditor().getTargetObjectId());
+        if (optionalSelf.isPresent()) {
+            Object formDescription = optionalSelf.get();
+            if (formDescription instanceof FormDescription) {
+                Optional<Object> widgetToRemove = this.objectService.getObject(editingContext, widgetId);
+                if (widgetToRemove.isPresent()) {
+                    return ((FormDescription) formDescription).getWidgets().remove(widgetToRemove.get());
+                }
+            }
+        }
+        return false;
```

Comment:
Wouldn't this code be better:

```
this.objectService.getObject(editingContextId, widgetId).ifPresent(this.editService::delete);
```

I think that your code could end up leaving the object as dangling?